include/revng/Model/Types.h

```diff
@@ -0,0 +1,335 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <algorithm>
+#include <optional>
+
+#include "llvm/ADT/StringRef.h"
+
+#include "revng/ADT/SortedVector.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/Debug.h"
+
+namespace flatc {
+
+struct Type {
+  enum TypeKind {
+    TK_Invalid,
+    TK_Primitive,
+    TK_Enum,
+    TK_Typedef,
+    TK_Struct,
+    TK_Union,
+    TK_FunctionPtr,
+  };
```

Comment:
Where's is this enum serialized?
For `enum`s, please adopt the pattern used [elsewhere](https://github.com/revng/revng/blob/develop/include/revng/Model/Binary.h#L546):
* `namespace`
* `enum Values`
* Specialize `llvm::yaml::ScalarEnumerationTraits`
* No prefix

Reply:
From a previous in-person discussion I had already did points 1, 2, 4. I just updated the pull request with these changes.

Point 3 is not strictly necessary, but I noticed that you have done so for all the equivalent enums in `Binary.h`. Is there a specific reason why you are doing so and not relying on llvm's yaml stuff to do it for you?

Reply:
3) what do you mean? There's no automagic YAML salsa for enums, you have to explicitly specialize those.

Reply:
Well, no, my bad. I've already done point 3 as well. That's why it worked.

All you asked in this comment is done. I'm resolving this conversation.